llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -607,176 +781,9 @@ Instruction *InstCombinerImpl::visitFMul(BinaryOperator &I) {
   if (Value *V = SimplifySelectsFeedingBinaryOp(I, Op0, Op1))
     return replaceInstUsesWith(I, V);
 
-  if (I.hasAllowReassoc()) {
```

Comment:
Can you pre-commit moving all of this code to a helper function in an NFC commit? It's hard to see the real diff here

Reply:
Thanks @arsenm for your idea,  I already split it into 2 commit, so do you mean it is better to try them with 2 independent PRs ?

Reply:
Oh right, I'm still getting confused by the PR interface. I would just pre-push the first commit

Reply:
Done, raise [PR71493](https://github.com/llvm/llvm-project/pull/71493) to precommit the first commit.
world you help me to check that, then I'll rebase this commit, thanks.

Reply:
Thanks very much, now it is clear, and I'll try address your previous commit.